app/models/user.rb

```diff
@@ -24,18 +24,23 @@ def pending_friends
   end
 
   # Users who have requested to be friends
+  # also doubles as pending_requests
   def friend_requests
```

Comment:
- [ ] `friend_requests` can be defined as association with SQL query instead of processing objects in application memory (as the database was created to store and query data it is always more efficient than ruby)
```
# Users who requested to be friends (needed for notifications)
has_many :inverted_friendships, -> { where confirmed: false }, class_name: "Friendship", foreign_key: "friend_id"
has_many :friend_requests, through: :inverted_friendships
```

Reply:
done